fastapi/app/main.py

```diff
@@ -0,0 +1,231 @@
+"""Module providing SpiceQL endpoints"""
+
+from ast import literal_eval
+from typing import Annotated, Any, Optional
+from fastapi import FastAPI, Query
+from pydantic import BaseModel, Field
+import pyspiceql
+
+
+# Models
+class MessageItem(BaseModel):
+    message: str
+
+class BodyModel(BaseModel):
+    result: Optional[Any] = None
+    error: Optional[str] = None
+
+class ResultModel(BodyModel):
+    result: Any = Field(serialization_alias='return')
+
+class ErrorModel(BodyModel):
+    error: str
+
+class ResponseModel(BaseModel):
+    statusCode: int
+    body: dict | str
+
+# Create FastAPI instance
+app = FastAPI()
+
+# General endpoints
+@app.get("/")
+async def root():
+    return {"message": "Visit the /docs endpoint to see the Swagger UI."}
+
+@app.post("/customMessage")
+async def message(
+    message_item: MessageItem
+    ):
+    return {"message": message_item.message}
+
+
+# SpiceQL endpoints
+@app.get("/getTargetStates")
+async def getTargetStates(
+    ets: Annotated[list[float], Query()] | str,
+    target: str,
+    observer: str,
+    frame: str,
+    abcorr: str,
+    mission: str,
+    ckQuality: str = "",
+    spkQuality: str = "",
+    searchKernels: bool = False):
+    try:
+        if isinstance(ets, str):
+            ets = literal_eval(ets)
+        result = pyspiceql.getTargetStates(ets, target, observer, frame, abcorr, mission, ckQuality, spkQuality, searchKernels)
+        body = ResultModel(result=result).model_dump(by_alias=True, exclude_none=True)
+        return ResponseModel(statusCode=200, body=body)
+    except Exception as e:
+        body = ErrorModel(error=str(e)).model_dump(exclude_none=True)
+        return ResponseModel(statusCode=500, body=body)
```

Comment:
This is pretty sweet, The nested models building out the response. No comment here, just a thanks for sharing.

Reply:
Actually did a little more refactoring to simplify the model structure and the serialization process, wouldn't have found this if you didn't bring this up, thanks! 

I think past-me was initially trying to turn `ResultModel` and `ErrorModel` as sub-models of a parent one so that `ResponseModel` could accept the base `BodyModel` and either types can be passed in. I was overcomplicating things (I think I was thinking in terms of Java bc OOP) when I just realized that I can do away with the `BodyModel` and set the `body` attribute of `ResponseModel` to accept either models. I wouldn't have to do a `model_dump()` of each body result to turn into a string or dict either 🤦‍♀️